bucoffea/vbfhinv/vbfhinvProcessor.py

```diff
@@ -174,118 +182,141 @@ def process(self, df):
-        selection.add('hemisphere', hemisphere)
-        selection.add('leadak4_id',leadak4_id.any())
-        selection.add('trailak4_id',trailak4_id.any())
-        selection.add('mjj', df['mjj'] > cfg.SELECTION.SIGNAL.DIJET.SHAPE_BASED.MASS)
-        selection.add('dphijj', df['dphijj'] < cfg.SELECTION.SIGNAL.DIJET.SHAPE_BASED.DPHI)
-        selection.add('detajj', df['detajj'] > cfg.SELECTION.SIGNAL.DIJET.SHAPE_BASED.DETA)
-
-        # Divide into three categories for trigger study
-        if cfg.RUN.TRIGGER_STUDY:
-            two_central_jets = (np.abs(diak4.i0.eta) <= 2.4) & (np.abs(diak4.i1.eta) <= 2.4)
-            two_forward_jets = (np.abs(diak4.i0.eta) > 2.4) & (np.abs(diak4.i1.eta) > 2.4)
-            one_jet_forward_one_jet_central = (~two_central_jets) & (~two_forward_jets)
-            selection.add('two_central_jets', two_central_jets.any())
-            selection.add('two_forward_jets', two_forward_jets.any())
-            selection.add('one_jet_forward_one_jet_central', one_jet_forward_one_jet_central.any())
 
         # Dimuon CR
         leadmuon_index=muons.pt.argmax()
-        selection.add('at_least_one_tight_mu', df['is_tight_muon'].any())
-        selection.add('dimuon_mass', ((dimuons.mass > cfg.SELECTION.CONTROL.DOUBLEMU.MASS.MIN) \
-                                    & (dimuons.mass < cfg.SELECTION.CONTROL.DOUBLEMU.MASS.MAX)).any())
-        selection.add('dimuon_charge', (dimuon_charge==0).any())
-        selection.add('two_muons', muons.counts==2)
-
-        # Single muon CR
-        selection.add('one_muon', muons.counts==1)
-        selection.add('mt_mu', df['MT_mu'] < cfg.SELECTION.CONTROL.SINGLEMU.MT)
 
         # Diele CR
         leadelectron_index=electrons.pt.argmax()
 
-
-        selection.add('one_electron', electrons.counts==1)
-        selection.add('two_electrons', electrons.counts==2)
-        selection.add('at_least_one_tight_el', df['is_tight_electron'].any())
-
-        selection.add('dielectron_mass', ((dielectrons.mass > cfg.SELECTION.CONTROL.DOUBLEEL.MASS.MIN)  \
-                                        & (dielectrons.mass < cfg.SELECTION.CONTROL.DOUBLEEL.MASS.MAX)).any())
-        selection.add('dielectron_charge', (dielectron_charge==0).any())
-        selection.add('two_electrons', electrons.counts==2)
-
-        # Single Ele CR
-        selection.add('met_el', met_pt > cfg.SELECTION.CONTROL.SINGLEEL.MET)
-        selection.add('mt_el', df['MT_el'] < cfg.SELECTION.CONTROL.SINGLEEL.MT)
-
         # Photon CR
         leadphoton_index=photons.pt.argmax()
 
         df['is_tight_photon'] = photons.mediumId \
                          & (photons.abseta < cfg.PHOTON.CUTS.TIGHT.ETA)
+      
+        # Process for each JES/JER variation,
+        # add common selections to each selection packer
+        for var in self._variations:
+            # Get the correct objects/quantities for each variation
+            selection = vmap.get_selection_packer(var)
```

Comment:
You already renamed the cuts, e.g.
`mjj` -> `mjj_{var}`

In that case, why do we need separate selection objects? we can store all cuts in the same selection object. That would also allow us to only do define e.g. the muon cuts one outside the loop rather than redoing them every time

Reply:
The selection objects can hold up to 64 selections (maybe there is some way around this but I couldn't figure it out). So as I try to loop through and fill a single selection object, it is overloaded at some point. That's why I decided to go with one selection-packer for each variation.

Reply:
Check this out: https://github.com/CoffeaTeam/coffea/blob/master/coffea/processor/helpers.py#L152-L158

Reply:
Alternative implementation of selections is in this commit: https://github.com/alpakpinar/bucoffea/commit/02c6f0847d10589f09368ee0f00c105ae8faece7

Still multiple selection objects are used, but common selections are added only once, and then copied for all other selection objects.